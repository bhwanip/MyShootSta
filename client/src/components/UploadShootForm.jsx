import React from 'react';
import TextInput from './TextInput';
import FileInput from './FileInput';

const UploadShootForm = (props) => (
    <div className="container">
        <div className=".col-md-offset-4 media-list">
            <div className="panel panel-default">
                <div className="panel-heading">
                    <h2 className="panel-title text-center">
                        Upload Video
                    </h2>
                </div>
                <div className="panel-body">
                    {(!!props.message) ? (<div className="alert alert-success">{props.message}</div>) : null}
                    <form onChange={props.hanleFormChange} onSubmit={props.handleSubmit} encType="multipart/form-data">
                    
                    <TextInput label="Title" placeholder="Enter the title..." name="title" value={props.title} {...props}/>
                    <TextInput label="Editor Name" placeholder="Enter the Editor Name..." name="editor" value={props.editor}  {...props}/>
                    <FileInput label="Upload Your Video..." name="shootVideo" {...props}/>
                    
                    <button className="btn btn-primary btn-lg" type="submit" >
                            Submit</button>
                    </form>
                </div>
            </div>
        </div>
    </div>
)

export default UploadShootForm;