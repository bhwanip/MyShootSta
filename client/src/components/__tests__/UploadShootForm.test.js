import React from 'react';
import renderer from 'react-test-renderer';

import UploadShootForm from '../UploadShootForm';

describe('<UploadShootForm/>', () => {
  it('should render correctly with success message', () => {

    let props = {
        message: 'Your Video has been uploaded sucessfully',
        hanleFormChange: jest.fn(),
        handleSubmit: jest.fn(),
        title: "TestTitle",
        editor: "TestEditor",
        label: "TestLabel",
        name: "TestName",
        handleTextChange: jest.fn(),
        captureFileEl: jest.fn(),
        handleFileChange: jest.fn()
    }

    let tree = renderer.create(<UploadShootForm {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with error message', () => {

    let props = {
        message: 'Error, please try again..',
        hanleFormChange: jest.fn(),
        handleSubmit: jest.fn(),
        title: "TestTitle",
        editor: "TestEditor",
        label: "TestLabel",
        name: "TestName",
        handleTextChange: jest.fn(),
        captureFileEl: jest.fn(),
        handleFileChange: jest.fn()
    }

    let tree = renderer.create(<UploadShootForm {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with no message', () => {

    let props = {
        message: '',
        hanleFormChange: jest.fn(),
        handleSubmit: jest.fn(),
        title: "TestTitle",
        editor: "TestEditor",
        label: "TestLabel",
        name: "TestName",
        handleTextChange: jest.fn(),
        captureFileEl: jest.fn(),
        handleFileChange: jest.fn()
    }

    let tree = renderer.create(<UploadShootForm {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});