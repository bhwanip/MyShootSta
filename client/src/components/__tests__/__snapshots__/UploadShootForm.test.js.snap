// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadShootForm/> should render correctly with error message 1`] = `
<div
  className="container"
>
  <div
    className=".col-md-offset-4 media-list"
  >
    <div
      className="panel panel-default"
    >
      <div
        className="panel-heading"
      >
        <h2
          className="panel-title text-center"
        >
          Upload Video
        </h2>
      </div>
      <div
        className="panel-body"
      >
        <div
          className="alert alert-success"
        >
          Error, please try again..
        </div>
        <form
          encType="multipart/form-data"
          onChange={[MockFunction]}
          onSubmit={[MockFunction]}
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="TestName"
            >
              TestLabel
            </label>
            <input
              className="form-control"
              name="TestName"
              onChange={[MockFunction]}
              placeholder="Enter the title..."
              type="text"
              value="TestTitle"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="TestName"
            >
              TestLabel
            </label>
            <input
              className="form-control"
              name="TestName"
              onChange={[MockFunction]}
              placeholder="Enter the Editor Name..."
              type="text"
              value="TestEditor"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="TestName"
            >
              TestLabel
            </label>
             
            <br />
            <input
              name="TestName"
              onChange={[MockFunction]}
              type="file"
            />
          </div>
          <button
            className="btn btn-primary btn-lg"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`<UploadShootForm/> should render correctly with no message 1`] = `
<div
  className="container"
>
  <div
    className=".col-md-offset-4 media-list"
  >
    <div
      className="panel panel-default"
    >
      <div
        className="panel-heading"
      >
        <h2
          className="panel-title text-center"
        >
          Upload Video
        </h2>
      </div>
      <div
        className="panel-body"
      >
        <form
          encType="multipart/form-data"
          onChange={[MockFunction]}
          onSubmit={[MockFunction]}
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="TestName"
            >
              TestLabel
            </label>
            <input
              className="form-control"
              name="TestName"
              onChange={[MockFunction]}
              placeholder="Enter the title..."
              type="text"
              value="TestTitle"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="TestName"
            >
              TestLabel
            </label>
            <input
              className="form-control"
              name="TestName"
              onChange={[MockFunction]}
              placeholder="Enter the Editor Name..."
              type="text"
              value="TestEditor"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="TestName"
            >
              TestLabel
            </label>
             
            <br />
            <input
              name="TestName"
              onChange={[MockFunction]}
              type="file"
            />
          </div>
          <button
            className="btn btn-primary btn-lg"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`<UploadShootForm/> should render correctly with success message 1`] = `
<div
  className="container"
>
  <div
    className=".col-md-offset-4 media-list"
  >
    <div
      className="panel panel-default"
    >
      <div
        className="panel-heading"
      >
        <h2
          className="panel-title text-center"
        >
          Upload Video
        </h2>
      </div>
      <div
        className="panel-body"
      >
        <div
          className="alert alert-success"
        >
          Your Video has been uploaded sucessfully
        </div>
        <form
          encType="multipart/form-data"
          onChange={[MockFunction]}
          onSubmit={[MockFunction]}
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="TestName"
            >
              TestLabel
            </label>
            <input
              className="form-control"
              name="TestName"
              onChange={[MockFunction]}
              placeholder="Enter the title..."
              type="text"
              value="TestTitle"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="TestName"
            >
              TestLabel
            </label>
            <input
              className="form-control"
              name="TestName"
              onChange={[MockFunction]}
              placeholder="Enter the Editor Name..."
              type="text"
              value="TestEditor"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="TestName"
            >
              TestLabel
            </label>
             
            <br />
            <input
              name="TestName"
              onChange={[MockFunction]}
              type="file"
            />
          </div>
          <button
            className="btn btn-primary btn-lg"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
